File: leetcode-submissions/1769_Minimum_Number_of_Operations_to_Move_All_Balls_to_Each_Box.ts
function minOperations(boxes: string): number[] {
    const set:Set<number>=new Set();
    for(let i:number=0;i<boxes.length;i++){
        if(boxes[i]==="1") set.add(i);
    }
    const ans:number[]=Array.from({length:boxes.length},()=>0);
    for(let i:number =0;i<boxes.length;i++){
        let moves :number=0;
        for(const j of set){
            moves+=Math.abs(i-j);
        }
        ans[i]=moves;
    }
    return ans ;
};

File: leetcode-submissions/1769_Minimum_Number_of_Operations_to_Move_All_Balls_to_Each_Box.cpp
class Solution {
public:
    vector<int> minOperations(string boxes) {
        unordered_set<int>s;
        for(int i=0;i<boxes.size();i++){
            if(boxes[i]=='1'){ s.insert(i);}
        }
        vector<int>ans(boxes.size(),0);
        for(int i=0;i<boxes.size();i++){
            int moves=0;
            for(auto index:s){
                moves+=abs(i-index);
            }
            ans[i]=moves;
        }
        return ans ;
    }
};